runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java

```diff
@@ -1031,17 +1031,19 @@ private void doPartitionAssignment(
                     int partitionIndex = 0;
                     int newPartitionPerTopic = numberOfPartitionsPerMember + extraPartition;
 
+                    IntHashSet.IntIterator iterator = p.iterator();
+
                     for (String member : members.keySet())
                     {
                         String consumerId = members.get(member);
                         List<TopicPartition> topicPartitions = consumers.computeIfAbsent(
                             member, tp -> new ArrayList<>());
                         IntHashSet partitions = new IntHashSet();
 
-                        IntHashSet.IntIterator iterator = p.iterator();
                         for (; partitionIndex < newPartitionPerTopic; partitionIndex++)
                         {
-                            partitions.add(iterator.next());
+                            final Integer partitionId = iterator.next();
```

Comment:
Please use `.nextValue()` instead of `.next()` on `IntIterator` to avoid boxing to `Integer`.